runtime/src/commonMain/kotlin/pbandk/ExtendableMessage.kt

```diff
@@ -0,0 +1,23 @@
+package pbandk
+
+@Suppress("UNCHECKED_CAST")
+interface ExtendableMessage : Message {
+    val extensionFields: pbandk.AtomicReference<Map<Int, Any>>
+    fun <M: Message, T> getExtension(fd: FieldDescriptor<M, T>): T {
+        var value = extensionFields.get()[fd.number] as? T
+        if (value != null) {
+            return value
+        } else {
+            value = unknownFields[fd.number]?.let { unknownField ->
+                unknownField.value.parseUnknownField(fd)
+            }
+        }
+        return if (value != null) {
+            extensionFields.set(extensionFields.get() + (fd.number to value as Any))
+            value
+        } else {
+            fd.type.defaultValue as T
```

Comment:
Following on from my comment in `CodeGenerator`, I realized that this default value code is not quite right. If `fd.type.hasPresence` is true, then the default value should be `null`. Only if `value == null` _and_ the `hasPresence` is false should we return `fd.type.defaultValue`.

You could also argue that rather than making the fix here, perhaps the `fd.type.defaultValue` property should be fixed to return `null` when `fd.type.hasPresence` is true...